propagation.go

```diff
@@ -0,0 +1,133 @@
+package opentracing
+
+import "errors"
+
+///////////////////////////////////////////////////////////////////////////////
+// CORE PROPAGATION INTERFACES:
+///////////////////////////////////////////////////////////////////////////////
+
+var (
+	// TraceNotFound occurs when the `carrier` passed to Extractor.JoinTrace()
+	// is valid and uncorrupted but has insufficient information to join or
+	// resume a trace.
+	TraceNotFound = errors.New("opentracing: Trace not found in Extraction carrier")
+
+	// InvalidCarrier errors occur when Injector.InjectSpan() or
+	// Extractor.JoinTrace() implementations expect a different type of
+	// `carrier` than they are given.
+	InvalidCarrier = errors.New("opentracing: Invalid Injection/Extraction carrier")
+
+	// TraceCorrupted occurs when the `carrier` passed to Extractor.JoinTrace()
+	// is of the expected type but is corrupted.
+	TraceCorrupted = errors.New("opentracing: Trace data corrupted in Extraction carrier")
+)
+
+// Injector is responsible for injecting Span instances in a manner suitable
+// for propagation via a format-specific "carrier" object. Typically the
+// injection will take place across an RPC boundary, but message queues and
+// other IPC mechanisms are also reasonable places to use an Injector.
+//
+// See Extractor and Tracer.Injector.
+type Injector interface {
+	// InjectSpan takes `span` and injects it into `carrier`. The actual type
+	// of `carrier` depends on the `format` passed to `Tracer.Injector()`.
+	//
+	// Implementations may return opentracing.InvalidCarrier or any other
+	// implementation-specific error if injection fails.
+	InjectSpan(span Span, carrier interface{}) error
```

Comment:
Please scrutinize this comment and the one for `JoinTrace` below... and `Tracer.Injector()/Extractor()`, too.